### PR TITLE
Fix DeprecationWarning: datetime.datetime.utcnow()

### PR DESCRIPTION
**Pull request by cclauss**
### Fix DeprecationWarning: datetime.datetime.utcnow() on Python 3.12
```
lib/python3.12/site-packages/celery/app/base.py:940: DeprecationWarning: datetime.datetime.utcnow()
  is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent
  datetimes in UTC: datetime.datetime.now(datetime.UTC).
    now_in_utc = to_utc(datetime.utcnow())
```
https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
you can make it draft before the builds are passing

File: t/unit/app/test_beat.py [hunk 1]
```diff
@@ -863,17 +863,17 @@ class test_schedule:
...
         x.utc_enabled = False
-        d2 = x.maybe_make_aware(datetime.utcnow())
+        d2 = x.maybe_make_aware(datetime.now(timezone.utc))
         assert d2.tzinfo
 
     def test_to_local(self):
         x = schedule(10, app=self.app)
         x.utc_enabled = True
-        d = x.to_local(datetime.utcnow())
+        d = x.to_local(datetime.utcnow())  # datetime.utcnow() is deprecated in Python 3.12
```
Comment (auvipy) [inline t/unit/app/test_beat.py:875]: shouldn't we change this as well?

Comment (cclauss) [reply t/unit/app/test_beat.py:875]: Tests will fail.

**Review by thedrow — APPROVED** [review]

**Comment by Nusnus** [comment]
Restarting CI, unrelated error

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
great job! thanks a lot

**Comment by cclauss** [comment]
There is more `utcnow()` to deal with but these instance are harder to figure out...
* https://github.com/search?q=repo%3Acelery%2Fcelery%20utcnow&type=code

**Pull request merged**